src/model/Patient.ts

```diff
@@ -118,43 +166,174 @@ export default class Patient implements IPatient {
   get fullNameDisplay(): string {
     let names: Array<string> = [];
     if (this.firstName && this.firstName.length > 0) names.push(this.firstName);
-    if (this.lastName != null && this.lastName.length > 0) names.push(this.lastName);
+    if (this.lastName != null && this.lastName.length > 0)
+      names.push(this.lastName);
     return names.join(" ");
   }
 
+  get addressText(): string {
+    if (!this.address || !this.address.length) return "";
+    return this.address.find((a) => a.use === "home")?.text;
+  }
+
+  set addressText(text: string) {
+    if (this.address == null) this.address = [];
+    let address: IAddress = this.address.find((a) => a.use === "home");
+    if (address == null) {
+      address = { use: "home" } as IAddress;
+      this.address.push(address);
+    }
+    address.text = text;
+  }
+
   get homeZip(): string {
     return this.address != null && this.address.length > 0
-        ? this.address.find((a) =>
-            a.use === "home")?.postalCode : null;
+      ? this.address.find((a) => a.use === "home")?.postalCode
+      : null;
   }
 
   set homeZip(text: string) {
     if (this.address == null) this.address = [];
-    let address: IAddress =
-        this.address.find((a) => a.use === "home");
+    let address: IAddress = this.address.find((a) => a.use === "home");
     if (address == null) {
-      address = {use: "home"} as IAddress;
+      address = { use: "home" } as IAddress;
       this.address.push(address);
     }
     address.postalCode = text;
   }
 
   get secondZip(): string {
     return this.address != null && this.address.length > 0
-        ? this.address.find((a) =>
-            a.use !== "home")?.postalCode : null;
+      ? this.address.find((a) => a.use !== "home")?.postalCode
+      : null;
   }
 
   set secondZip(text: string) {
     if (this.address == null) this.address = [];
-    let address: IAddress =
-        this.address.find((a) => a.use !== "home");
+    let address: IAddress = this.address.find((a) => a.use !== "home");
     if (address == null) {
       address = {} as IAddress;
       this.address.push(address);
     }
     address.postalCode = text;
   }
+
+  addContact(
+    name: string = null,
+    phoneNumber: string = null,
+    email: string = null
+  ) {
+    if (!(name || phoneNumber || email)) return;
+    if (this.contact == null) this.contact = [];
+    let contactToAdd: IPatient_Contact = {} as IPatient_Contact;
+    if (name) {
+      contactToAdd.name = {
+        text: name,
+      } as IHumanName;
+    }
+    if (phoneNumber) {
+      contactToAdd.telecom = [];
+      contactToAdd.telecom.push({
+        system: ContactPointSystemKind._phone,
+        value: phoneNumber,
+      });
+    }
+    if (email) {
+      if (!contactToAdd.telecom) contactToAdd.telecom = [];
+      contactToAdd.telecom.push({
+        system: ContactPointSystemKind._email,
+        value: email,
+      });
+    }
+    this.contact.push(contactToAdd);
```

Comment:
Can you please also set contact.relationship to `EP` per https://www.hl7.org/fhir/valueset-patient-contactrelationship.html ?

Reply:
As I follow the breadcrumbs from my other link to https://terminology.hl7.org/5.1.0/CodeSystem-v2-0131.html#v2-0131-EP, I think `coding` should be `C` (`EC` deprecated) and probably best to include `display` of `Emergency Contact` for transparency. Does that make sense, or am I overlooking something you found?

Reply:
thanks @mcjustin I did make the fix, e.g.
```
{
      "relationship": [
        {
          "coding": [
            {
              "system": "http://hl7.org/fhir/ValueSet/patient-contactrelationship",
              "code": "C",
              "display": "Emergency Contact"
            }
          ]
        }
      ],
      "name": {
        "text": "Test Test"
      },
      "telecom": [
        {
          "system": "phone",
          "value": "1231231111"
        },
        {
          "system": "email",
          "value": "achen2401+test3@gmail.com"
        }
      ]
    }
```